buildSrc/src/main/kotlin/org/gradle/plugins/buildscan/BuildScanConfigurationPlugin.kt

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.plugins.buildscan
+
+import com.gradle.scan.plugin.BuildScanExtension
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.plugins.quality.Checkstyle
+import org.gradle.api.plugins.quality.CodeNarc
+import org.gradle.api.reporting.ReportContainer
+import org.gradle.api.reporting.Reporting
+import org.gradle.kotlin.dsl.the
+import org.jsoup.Jsoup
+import org.jsoup.parser.Parser
+import org.gradle.api.internal.GradleInternal
+import org.gradle.internal.classloader.ClassLoaderHierarchyHasher
+import kotlin.concurrent.thread
+
+
+class BuildScanConfigurationPlugin : Plugin<Project> {
+    private
+    val isCiServer = System.getenv().containsKey("CI")
+
+    override fun apply(project: Project): Unit = project.run {
+        apply {
+            plugin("com.gradle.build-scan")
```

Comment:
Why don't we apply the plugin by type, given that we now have the plugin on the classpath?

Reply:
So far out pattern is to always apply them by id. Something for @rodrigo and @paul to comment on. You would get content assist but would pay with higher coupling to an implementation detail.